src/spikeinterface/sorters/external/kilosortbase.py

```diff
@@ -215,16 +215,23 @@ def _run_from_folder(cls, sorter_output_folder, params, verbose):
             raise Exception(f"{cls.sorter_name} returned a non-zero exit code")
 
         # Clean-up temporary files
-        if params["delete_tmp_files"]:
-            for temp_file in sorter_output_folder.glob("*.m"):
-                temp_file.unlink()
-            for temp_file in sorter_output_folder.glob("*.mat"):
-                temp_file.unlink()
-            if (sorter_output_folder / "temp_wh.dat").exists():
-                (sorter_output_folder / "temp_wh.dat").unlink()
         if params["delete_recording_dat"] and (recording_file := sorter_output_folder / "recording.dat").exists():
             recording_file.unlink()
 
+        if params["delete_tmp_files"]:
```

Comment:
```suggestion
        if isinstance(params["delete_tmp_files"], bool):
            if params["delete_tmp_files"]:
                tmp_to_remove = ("matlab_files", "temp_wh.dat")
            else:
                tmp_to_remove = ()
        else:
            assert isinstance(params["delete_tmp_files"], (tuple, list)), "..."
       
       if "temp_wh.dat" in tmp_to_remove: ... 
```
@JoeZiminski I think it would be easier to read if we test for boolean first!